tests/components/fritz/test_config_flow.py

```diff
@@ -0,0 +1,293 @@
+MOCK_USER_DATA_HOST = MOCK_CONFIG_HOST[DOMAIN][CONF_DEVICES][0]
+MOCK_DEVICE_INFO = {
+    ATTR_HOST: MOCK_HOST,
+    ATTR_NEW_SERIAL_NUMBER: MOCK_SERIAL_NUMBER,
+}
+MOCK_SSDP_DATA = {
+    ATTR_SSDP_LOCATION: "https://fake_host:12345/test",
+    ATTR_UPNP_FRIENDLY_NAME: "fake_name",
+    ATTR_UPNP_UDN: "uuid:only-a-test",
+}
+
+
+@pytest.fixture()
+def fc_class_mock(mocker):
+    """Fixture that sets up a mocked FritzConnection class."""
+    result = mocker.patch("fritzconnection.FritzConnection", autospec=True)
+    result.return_value = FritzConnectionMock()
+    yield result
+
+
+async def test_user(hass: HomeAssistantType, fc_class_mock):
+    """Test starting a flow by user."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "start_config"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["data"][CONF_HOST] == "fake_host"
+        assert result["data"][CONF_PASSWORD] == "fake_pass"
+        assert result["data"][CONF_USERNAME] == "fake_user"
+        assert not result["result"].unique_id
+
+
+async def test_user_already_configured(hass: HomeAssistantType, fc_class_mock):
+    """Test starting a flow by user."""
+
+    mock_config = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "start_config"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "start_config"
+
+
+async def test_exception_1(hass: HomeAssistantType):
+    """Test starting a flow by user."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=FritzSecurityError,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "start_config"
+
```

Comment:
Move this outside of the patch since the patched code is not being called in the first step

Reply:
Fixed